src/shipping/components/ShippingZoneInfo/ShippingZoneInfo.tsx

```diff
@@ -1,30 +1,71 @@
 import Card from "@material-ui/core/Card";
 import CardContent from "@material-ui/core/CardContent";
+import { makeStyles } from "@material-ui/core/styles";
 import TextField from "@material-ui/core/TextField";
+import CardSpacer from "@saleor/components/CardSpacer";
 import CardTitle from "@saleor/components/CardTitle";
 import { ShippingErrorFragment } from "@saleor/fragments/types/ShippingErrorFragment";
 import { commonMessages } from "@saleor/intl";
 import { getFormErrors } from "@saleor/utils/errors";
 import getShippingErrorMessage from "@saleor/utils/errors/shipping";
 import React from "react";
+import { defineMessages } from "react-intl";
 import { useIntl } from "react-intl";
 
+const messages = defineMessages({
+  maxDays: {
+    defaultMessage: "Max Delivery Time",
+    description: "label"
+  },
+  minDays: {
+    defaultMessage: "Min Delivery Time",
+    description: "label"
+  },
+  name: {
+    defaultMessage: "Shipping rate name",
+    description: "label"
+  }
+});
+
+const useStyles = makeStyles(
+  theme => ({
+    deliveryTimeFields: {
+      display: "grid",
+      gridColumnGap: theme.spacing(2),
+      gridRowGap: theme.spacing(1),
+      gridTemplateColumns: "1fr 1fr 1fr",
+      [theme.breakpoints.down("md")]: {
+        gridTemplateColumns: "1fr 1fr"
+      },
+      [theme.breakpoints.down("xs")]: {
+        gridTemplateColumns: "1fr"
+      }
+    }
+  }),
+  { name: "ShippingZoneInfo" }
+);
+
+interface ShippingZoneData {
+  name: string;
+  minDays?: string;
+  maxDays?: string;
+}
+
 export interface ShippingZoneInfoProps {
-  data: Record<"name", string>;
+  data: ShippingZoneData;
+  hasDaysRange?: boolean;
   disabled: boolean;
   errors: ShippingErrorFragment[];
   onChange: (event: React.ChangeEvent<any>) => void;
 }
 
-const ShippingZoneInfo: React.FC<ShippingZoneInfoProps> = ({
-  data,
-  disabled,
-  errors,
-  onChange
-}) => {
+const ShippingZoneInfo: React.FC<ShippingZoneInfoProps> = props => {
+  const { data, hasDaysRange = false, disabled, errors, onChange } = props;
```

Comment:
Why not leave the destructuring inside the fn argument instead?

Reply:
Because `props` might be passed in `const classes = useStyles(props);` further.